src/index.ts

```diff
@@ -410,18 +392,25 @@ export function extractValueOrLiteral<
 export type Rule =
   | string
   | {
-    if: Rule
+    if: And | Or | string
     then: Rule
     else?: Rule
   }
   | {
-    and: Rule[]
+    and: And
   }
   | {
-    or: Rule[]
+    or: Or
   }
   | {
-    return: Rule
+    return: string | string[]
   }
   | { try: Rule, catch: Rule }
   | Rule[];
```

Comment:
I like where you're going here!

If we keep refining these types, I imagine the top-level rules could look like this:

```ts
export type Rule =
  | string
  | IfThenElseRule
  | AndRule
  | OrRule
  | ReturnRule
  | TryCatchRule
  | Rule[];

type RulesStrings = string | string[];
 
interface ReturnRule {
  return: RulesStrings
}

interface TryCatchRule { try: Rule, catch: Rule }

interface IfThenElseRule {
  if: AndRule | OrRule | string
  then: Rule
  else?: Rule
}
```